### PR TITLE
School Library #3: Create diferent kind of relationships between Classes

### PR DESCRIPTION
**Pull request by danifromecuador**
## In this pull request, I've:
- Created a `feature/setup_classes-objects_associations` branch
- Created `student.rb`, `classroom.rb`, `rental.rb`, and book.rb` files with their respective classes
- Created `has-many` /`belongs-to`relationship between `Classroom` and `Student`
- Created `many-to-many` relationship between `Person` and `Book` using the intermediate class`Rental`

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,12 +1,16 @@
 require_relative 'person'
+require_relative 'classroom'
 
 class Student < Person
-  def initialize(age, classroom = 'unknown', name = 'Unknown', permission: true)
```
Comment (t-yanick) [inline student.rb:4]: - [x] Great job working on this milestone. You are supposed to define a getter method for the instance variable @classroom within a class. The generated getter method allows you to access the value of the @classroom instance variable from outside the Student class.
`attr_reader :classroom`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (t-yanick) [inline book.rb]: - [x] You are supposed to have both Getters and Setters for @rentals
`attr_accessor :rentals`
The getter method allows you to access the value of the @rentals instance variable from outside the class, while the setter method allows you to assign a new value to the @rentals instance variable.

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (t-yanick) [inline rental.rb]: - [x] You are supposed to define Getters and Setters for @book and @person instance variables.
`attr_accessor :date, :book, :person`
The getter method allows you to access the value of the @book and @person instance variables from outside the class, while the setter method allows you to assign a new value to the @book and @person instance variables.

File: person.rb [hunk 4]
```diff
@@ -1,19 +1,22 @@
 require_relative 'nameable'
 require_relative 'capitalize_decorator'
 require_relative 'trimmer_decorator'
+require_relative 'rental'
+require_relative 'book'
 
 class Person
+  attr_reader :id, :rentals
```
Comment (t-yanick) [inline person.rb]: - [x] You are supposed to have Getters and Setters for @rentals
`attr_accessor :name, :age, :rentals`
The getter method allows you to access the value of the @rentals instance variable from outside the class, while the setter method allows you to assign a new value to the @rentals instance variable.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @danifromecuador , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All classes and methods present
:heavy_check_mark: All associations correctly setup
:heavy_check_mark: Linters all passing
:heavy_check_mark: Use of correct Gitflow
:heavy_check_mark: Good commit messages 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (danifromecuador) [reply student.rb:4]: Thanks, done!

File: book.rb (hunk 2 above)
Comment (danifromecuador) [reply book.rb]: Thanks, it's done

File: rental.rb (hunk 3 above)
Comment (danifromecuador) [reply rental.rb]: thanks, that's done!

File: person.rb (hunk 4 above)
Comment (danifromecuador) [reply person.rb]: Thanks, done.

**Comment by danifromecuador** [comment]
Dear @t-yanick all the required changes were done, Thanks to you for your detailed review.

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @danifromecuador , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All required changes implemented 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**